providers/Providers.tsx

```diff
@@ -12,15 +12,15 @@ import { useSearchParams, useRouter, usePathname } from 'next/navigation';
 import { Header } from '@/components/Header/Header';
 import { UmiProvider } from './UmiProvider';
 import { EnvProvider } from './EnvProvider';
-import { Env } from './useEnv';
+import { Env, envOptions } from './useEnv';
 
 export function Providers({ children }: { children: ReactNode }) {
   const router = useRouter();
   const searchParams = useSearchParams();
   const pathname = usePathname();
   const queryEnv = searchParams.get('env');
   const [client] = useState(new QueryClient());
-  const [env, setEnv] = useState<Env>((queryEnv === 'mainnet' || queryEnv === 'devnet') ? queryEnv : 'mainnet');
+  const [env, setEnv] = useState<Env>((queryEnv === 'mainnet' || queryEnv === 'devnet') ? queryEnv : envOptions[0]?.env);
```

Comment:
I think you found a bug here (thanks!), the logic should probably be: if the `queryEnv` is valid (i.e. contained within `envOptions`) then set the corresponding env otherwise use the first thing in the env options

I checked on the current site and when navigating to a page via url and env set as a search param, the env for sonic/eclipse is not set properly.